### PR TITLE
Support building a QAT onnx model using onnxblock

### PR DESCRIPTION
**Pull request by baijumeswani**
After performing static quantization, the quant params are a part of the model initializers.

`onnxblock` treats model initializers as model parameters (for most scenarios) since it does not know how to (*smartly*) distinguish between model parameters and model initializers (consts or overridable initializers).

Quant params are not learnable (for now), so these parameters should not be used to build gradients. Moreover, we even avoid moving these parameters to graph inputs because these parameters are simply scalars (or 1d tensors).

This pull request adds support for doing the above in `onnxblock`.

### CODE REVIEW

File: orttraining/orttraining/python/training/onnxblock/_graph_utils.py [hunk 1]
```diff
@@ -58,7 +61,7 @@ def build_gradient_graph(accessor, user_args_requiring_grad, user_args_not_requi
     graph_inputs = model.graph.input
     initializers = []
     for initializer in model.graph.initializer:
-        if not initializer.name.startswith("onnx::"):
+        if not initializer.name.startswith("onnx::") and initializer.name not in quant_params:
```
Comment (askhade) [inline orttraining/orttraining/python/training/onnxblock/_graph_utils.py:64]: what does onnx:: refer to?

Comment (baijumeswani) [reply orttraining/orttraining/python/training/onnxblock/_graph_utils.py:64]: This is a hack (and not a very good one at that).
- To build the gradient graph, we need to know which onnx model initializer is a "pytorch model parameter" and which is constant initializers that the exporter introduced.
- This is needed primarily to know which initializers need their gradient computed.
- There is no good way of knowing this from the exported onnx model.
- The way I do it here is that I check if the exported onnx model has the prefix `onnx::`. If it does, then this was introduced by the exporter and hence is not a pytorch model parameter. So, no need to build gradient for this initializer.
- While this is a hack, I have requested a feature to the exporter team to add information in the onnx model to be able to tell which initializer is a model parameter and which is not.

**Review by askhade — APPROVED** [review]
